test/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java

```diff
@@ -155,4 +157,56 @@ public void checkBackpointersWork() {
         }
     }
 
+    @Test
+    public void SequencerWillResetTails() throws Exception {
+        UUID streamA = UUID.nameUUIDFromBytes("streamA".getBytes());
+        UUID streamB = UUID.nameUUIDFromBytes("streamB".getBytes());
+        UUID streamC = UUID.nameUUIDFromBytes("streamC".getBytes());
+
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamA))));
+        long tailA = getLastPayloadMessageAs(TokenResponse.class).getToken().getTokenValue();
+
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamB))));
+        long tailB = getLastPayloadMessageAs(TokenResponse.class).getToken().getTokenValue();
+
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamC))));
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamC))));
+
+        long tailC = getLastPayloadMessageAs(TokenResponse.class).getToken().getTokenValue();
```

Comment:
This just tests the reset part, can you also add tests where the sequencer is restarted and doesn't service token requests until it has been bootstrapped by the management server